File: frontend/src/types/application/ApplicationSearchDto.ts
import { ApplicationStatusDto } from '@/types/application/ApplicationStatusDto.ts'

export interface ApplicationSearchDto {
  id: number
  jobTitle: string
  company: string
  status: ApplicationStatusDto
  source: string | undefined
  description: string | undefined
  createdAt: string
  updatedAt: string | undefined
  isArchived: boolean
}

File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/application/dto/ApplicationSearchDto.kt
package ch.streckeisen.mycv.backend.application.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime

data class ApplicationSearchDto(
    val id: Long,
    val jobTitle: String,
    val company: String,
    val status: ApplicationStatusDto,
    val source: String?,
    val description: String?,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime?,
    @field:JsonProperty(value = "isArchived")
    val isArchived: Boolean
)

